packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -72,31 +126,103 @@ export async function forceLightningLwcPreview(sourceUri: vscode.Uri) {
     return;
   }
 
+  const platformSelection = await vscode.window.showQuickPick(platformInput, {
+    placeHolder: nls.localize('force_lightning_lwc_preview_platform_selection')
+  });
+  if (!platformSelection) {
+    console.log(`${logName}: No valid selection made for preview...`);
+    return;
+  }
+
   const fullUrl = `${DEV_SERVER_PREVIEW_ROUTE}/${componentName}`;
+  const desktopSelected = platformSelection.id === PreviewPlatformType.Desktop;
+  let target: string = platformSelection.defaultTargetName;
+  if (!desktopSelected) {
+    let placeholderText = nls.localize(
+      'force_lightning_lwc_preview_target_default'
+    );
+    const deviceConfig = getWorkspaceSettings().get('rememberDevice') || false;
+    const lastTarget = getRememberedDevice(platformSelection);
 
-  if (DevServerService.instance.isServerHandlerRegistered()) {
-    try {
-      await openBrowser(fullUrl);
-      telemetryService.sendCommandEvent(logName, startTime);
-    } catch (e) {
-      showError(e, logName, commandName);
+    // Remember device setting enabled and previous device retrieved.
+    if (deviceConfig && lastTarget) {
+      placeholderText = nls.localize(
+        'force_lightning_lwc_preview_target_remembered',
+        lastTarget
+      );
+      target = lastTarget;
     }
-  } else {
-    console.log(`${logName}: server was not running, starting...`);
-    const preconditionChecker = new SfdxWorkspaceChecker();
-    const parameterGatherer = new EmptyParametersGatherer();
-    const executor = new ForceLightningLwcStartExecutor({
-      openBrowser: true,
-      fullUrl
+    const targetName = await vscode.window.showInputBox({
+      placeHolder: placeholderText
     });
 
-    const commandlet = new SfdxCommandlet(
-      preconditionChecker,
-      parameterGatherer,
-      executor
-    );
+    if (targetName === undefined) {
+      vscode.window.showInformationMessage(
+        nls.localize('force_lightning_lwc_preview_device_cancelled')
+      );
+      return;
+    }
+
+    // New target device entered
+    if (targetName !== '') {
+      updateRememberedDevice(platformSelection, targetName);
```

Comment:
I don't immediately see any harm in saving the last used device, regardless of whether the user configures that behavior (which is what we're doing here).  But I figured I'd raise the question.

Reply:
That is a good call out.  I was thinking long term term instead of remembering the last device we way want to display a list of recently used devices (or not depending on the setting) and would always maintain the list.  But I don't feel strongly about this either way.